projects/driver_controls/src/ads1015.c

```diff
@@ -0,0 +1,151 @@
+#include <ads1015.h>
+#include <stdbool.h>
+#include <stdio.h>
+
+#include "ads1015_bitmasks.h"
+#include "delay.h"
+#include "gpio_it.h"
+#include "i2c.h"
+
+// Union definition to help process registers data
+typedef struct ADS1015Register {
+  uint8_t data[2];
+} ADS1015Register;
+
+// Since the ADS1015 stores data in two's complement, an int16_t will be used to hold
+// the raw readings
+typedef struct ADS1015Interrupt {
+  ADS1015Callback callback;
+  void *context;
+  int16_t reading;
+} ADS1015Interrupt;
+
+static I2CPort s_i2c_port;
+static ADS1015Channel s_current_channel;
+
+static ADS1015Interrupt s_interrupts[NUM_ADS1015_CHANNELS];
+
+static StatusCode prv_channel_valid(ADS1015Channel channel) {
```

Comment:
Titus and I have agreed that `prv` guard clauses while efficient are somewhat opaque and reduce readability. It would be preferable if you inline this.